src/compiler/utilities.ts

```diff
@@ -10338,3 +10338,8 @@ export function getPropertyNameFromType(type: StringLiteralType | NumberLiteralT
     }
     return Debug.fail();
 }
+
+/** @internal */
+export function isExpandoPropertyDeclarationCandidate(declaration: Declaration | undefined): declaration is PropertyAccessExpression | ElementAccessExpression | BinaryExpression {

```

Comment:
```suggestion
export function isExpandoPropertyDeclaration(declaration: Declaration | undefined): declaration is PropertyAccessExpression | ElementAccessExpression | BinaryExpression {
```

unless it's already taken, in which case I'd prefer the name `isExpandoPropertyDeclarationLocation`